test/640_proxy_restart_reattaches_test.sh

```diff
@@ -27,39 +22,41 @@ proxy_start_container_with_dns $HOST1 -di --name=c1 --restart=always
 C2=$(container_ip $HOST1 c2)
 
 proxy docker_on $HOST1 restart -t=1 c2
-check_attached
+wait_for_attached $HOST1 c2
+assert_dns_record $HOST1 c1 $NAME $C2
 
 # Restart weave router
 docker_on $HOST1 restart weave
+wait_for_attached $HOST1 c2
 sleep 1
-check_attached
+assert_dns_record $HOST1 c1 $NAME $C2
```

Comment:
We could eliminate the sleeps at a later date by implementing `wait_for_dns_record`.